crates/accelerate/src/twirling.rs

```diff
@@ -123,12 +128,59 @@ const CZ_MASK: u8 = 4;
 const ECR_MASK: u8 = 2;
 const ISWAP_MASK: u8 = 1;
 
+#[inline(always)]
+fn diff_frob_norm(array: ArrayView2<Complex64>, gate_matrix: ArrayView2<Complex64>) -> f64 {
+    let mut res: f64 = 0.;
+    for i in 0..4 {
+        for j in 0..4 {
+            let gate = gate_matrix[[i, j]];
+            let twirled = array[[i, j]];
+            let diff = twirled - gate;
+            res += (diff.conj() * diff).re;
+        }
+    }
+    res.sqrt()
+}
+
+fn generate_twirling_set(gate_matrix: ArrayView2<Complex64>) -> Vec<([StandardGate; 4], f64)> {
+    let mut out_vec = Vec::with_capacity(16);
+    let i_matrix = aview2(&ONE_QUBIT_IDENTITY);
+    let x_matrix = aview2(&qiskit_circuit::gate_matrix::X_GATE);
+    let y_matrix = aview2(&qiskit_circuit::gate_matrix::Y_GATE);
+    let z_matrix = aview2(&qiskit_circuit::gate_matrix::Z_GATE);
+    let iter_set = [
+        (IGate, i_matrix),
+        (XGate, x_matrix),
+        (YGate, y_matrix),
+        (ZGate, z_matrix),
+    ];
+    for (i, i_mat) in &iter_set {
+        for (j, j_mat) in &iter_set {
+            let before_matrix = kron(j_mat, i_mat);
+            let half_twirled_matrix = gate_matrix.dot(&before_matrix);
+            for (k, k_mat) in &iter_set {
+                for (l, l_mat) in &iter_set {
+                    let after_matrix = kron(l_mat, k_mat);
+                    let twirled_matrix = after_matrix.dot(&half_twirled_matrix);
+                    let norm: f64 = diff_frob_norm(twirled_matrix.view(), gate_matrix);
+                    if norm.abs() < 1e-15 {
+                        out_vec.push(([*i, *j, *k, *l], 0.));
+                    } else if (norm - 4.).abs() < 1e-15 {
+                        out_vec.push(([*i, *j, *k, *l], PI));
+                    }
```

Comment:
We don't actually need to use the proper Frobenius norm - we can use the square of the norm and avoid a call to `sqrt` (adjust the `norm - 4.` accordingly).

Really neither here nor there, but there's only four values `norm` could have that would satisfy the check on line 168 for that tolerance - `[4.-4.*eps, 4.-2.*eps, 4., 4.+4.*eps]` where `eps` is the float epsilon ($2^{-52}\approx 2.22\times 10^{-16}$).

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13331/commits/ca72bd0b8686e8c83d95bde1e1dc1fb7124fa895